pkg/operator/controller/dns/controller.go

```diff
@@ -0,0 +1,167 @@
+		if errors.IsNotFound(err) {
+			log.Info("dnsrecord not found; reconciliation will be skipped", "request", request)
+			return reconcile.Result{}, nil
+		} else {
+			return reconcile.Result{RequeueAfter: 15 * time.Second}, fmt.Errorf("failed to get dnsrecord: %v", err)
+		}
+	}
+
+	// If the DNS record was deleted, clean up and return.
+	if record.DeletionTimestamp != nil {
+		if err := r.delete(record); err != nil {
+			return reconcile.Result{RequeueAfter: 15 * time.Second}, err
+		}
+		return reconcile.Result{}, nil
+	}
+
+	ingressName, ok := record.Labels[manifests.OwningIngressControllerLabel]
+	if !ok {
+		return reconcile.Result{}, fmt.Errorf("missing owner label")
+	}
+	ingress := &operatorv1.IngressController{}
+	if err := r.cache.Get(context.TODO(), types.NamespacedName{Namespace: record.Namespace, Name: ingressName}, ingress); err != nil {
+		return reconcile.Result{}, fmt.Errorf("failed to get dnsrecord for record %s: %v", record.Name, err)
+	}
+
+	// TODO: Will soon need to honor scope of the ingresscontroller's LoadBalancer
+	// strategy.
+	zones := []configv1.DNSZone{}
+	if dnsConfig.Spec.PrivateZone != nil {
+		zones = append(zones, *dnsConfig.Spec.PrivateZone)
+	}
+	if dnsConfig.Spec.PublicZone != nil {
+		zones = append(zones, *dnsConfig.Spec.PublicZone)
+	}
+
+	// TODO: If we compare desired zones with zones from status (i.e. actual
+	// zones), we can filter to list to those zones which haven't already been
+	// successfully ensured. Then the DNS provider wouldn't need such aggressive
+	// internal caching.
+	result := reconcile.Result{}
+	statuses := []iov1.DNSZoneStatus{}
+	for i := range zones {
+		zone := zones[i]
+		conditions := []iov1.DNSZoneCondition{}
+		err := r.dnsProvider.Ensure(record, zone)
+		if err != nil {
+			conditions = append(conditions, iov1.DNSZoneCondition{
+				Type:               iov1.DNSRecordFailedConditionType,
+				Status:             string(operatorv1.ConditionTrue),
+				Reason:             "ProviderError",
+				Message:            fmt.Sprintf("The DNS provider failed to ensure the record: %v", err),
+				LastTransitionTime: metav1.Now(),
+			})
+			result.RequeueAfter = 30 * time.Second
+		}
+		statuses = append(statuses, iov1.DNSZoneStatus{
+			DNSZone:    zone,
+			Conditions: conditions,
+		})
+	}
+
+	// TODO: only update if status changed
+	updated := record.DeepCopy()
+	updated.Status.Zones = statuses
+	if err := r.client.Status().Update(context.TODO(), updated); err != nil {
+		return reconcile.Result{RequeueAfter: 10 * time.Second}, fmt.Errorf("failed to update dnsrecord status: %v", err)
+	}
+
+	return result, nil
+}
+
+func (r *reconciler) delete(record *iov1.DNSRecord) error {
+	var errs []error
+	for i := range record.Status.Zones {
+		zone := record.Status.Zones[i].DNSZone
+		err := r.dnsProvider.Delete(record, zone)
+		if err != nil {
+			errs = append(errs, err)
+		} else {
+			log.Info("provider deleted record %s in zone %v", record.Name, zone)
```

Comment:
I believe L154 should be something like:
```
log.Info("provider deleted record",  "name", record.Name, "zone", zone)
```

Reply:
Fixed